app/javascript/components/NamedAvatar.js

```diff
@@ -16,17 +20,79 @@ const styles = {
   name: {
     fontWeight: 600,
   },
-  subtitle: {},
+  remove: {
+    marginLeft: '20px',
+    cursor: 'pointer',
+  },
+  dialogActionsLink: {
+    padding: '15px',
+    cursor: 'pointer',
+    fontSize: '16px',
+  },
 }
 
-const NamedAvatar = props => (
+const dialogActions = (togglePopover, onRemove) => [
+  <a
+    style={styles.dialogActionsLink}
+    onClick={() => {
+      togglePopover('removeUserFromEvent')
+    }}
+  >
+    Cancel
+  </a>,
+  <a
+    style={styles.dialogActionsLink}
+    onClick={() => {
+      onRemove()
+      togglePopover('removeUserFromEvent')
+    }}
+  >
+    Delete
+  </a>,
+]
+
+const NamedAvatar = ({ userId, image, name, subtitle, showRemove, onRemove, togglePopover, popover }) => (
   <div style={styles.container}>
-    <Avatar {...props} />
+    <Avatar image={image} />
     <div style={styles.details}>
-      <span style={styles.name}>{props.name}</span>
-      <span style={styles.subtitle}>{props.subtitle || '\u00a0'}</span>
+      <span style={styles.name}>{name}</span>
+      <span style={styles.subtitle}>{subtitle || '\u00a0'}</span>
     </div>
+    {showRemove && (
+      <a
+        style={styles.remove}
+        onClick={() => {
+          togglePopover('removeUserFromEvent', { userId })
+        }}
+      >
+        Remove
+      </a>
+    )}
+    {present(popover) &&
+      popover.data.userId === userId && (
+        <Dialog
+          title={'Remove User ' + name + ' from Event'}
```

Comment:
can use template strings here 

```jsx
title={`Remove User ${name} from Event`}
```

I would also maybe just drop the word "User" from the thing so it reads "Remove hello from Event"